test/csrf.spec.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * @adonisjs/shield
+ *
+ * (c) ? (Please advice before merge, thanks!)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+*/
+
+import test from 'japa'
+import Tokens from 'csrf'
+import { csrf } from '../src/csrf'
+import { getCtxWithSession, getCtxFromIncomingMessage, getCsrfMiddlewareInstance } from '../test-helpers'
+
+const Csrf = new Tokens()
+const APPLICATION_SECRET_KEY = 'secret-key'
+
+test.group('Csrf', () => {
+  test('return noop function when enabled is false', async (assert) => {
+    const middlewareFn = csrf({ enabled: false }, APPLICATION_SECRET_KEY)
+    const ctx = await getCtxWithSession()
+    middlewareFn(ctx)
+
+    assert.isUndefined(ctx.request.csrfToken)
+  })
+
+  test('generate new, and valid csrf token for every new request', async (assert) => {
+    const csrfMiddleware = await getCsrfMiddlewareInstance({ enabled: true }, APPLICATION_SECRET_KEY)
+
+    const TEST_CSRF_SECRET = await csrfMiddleware.getCsrfSecret()
+    const TEST_CSRF_TOKEN = csrfMiddleware.generateCsrfToken(TEST_CSRF_SECRET)
+
+    const ctx = await getCtxFromIncomingMessage({
+      'x-csrf-token': TEST_CSRF_TOKEN,
+    })
+
+    ctx.session.put('csrf-secret', csrfMiddleware.session.get('csrf-secret'))
```

Comment:
What this line is intended to do?

Reply:
To test that the csrf token is correctly verified, first I created a token, and while creating a token, a new csrf secret is generated. Next, I share that secret with the second context I generated. That way they both have the same secret and the verification can go through.

Reply:
Okay, makes sense :)